kimchi/src/prover.rs

```diff
@@ -650,123 +688,191 @@ where
+                    f + lin
+                };
+
+                drop(env);
+                drop(lookup_sorted8);
+                drop(lookup_aggreg8);
+                drop(lookup_table_combined);
+
+                // see https://o1-labs.github.io/mina-book/crypto/plonk/maller_15.html#the-prover-side
+                f.chunk_polynomial(zeta_to_srs_len, index.max_poly_size)
             };
 
-            drop(env);
-            drop(lookup_sorted8);
-            drop(lookup_aggreg8);
-            drop(lookup_table_combined);
+            let t_chunked = quotient_poly.chunk_polynomial(zeta_to_srs_len, index.max_poly_size);
 
-            f.chunk_polynomial(zeta_to_srs_len, index.max_poly_size)
+            &f_chunked - &t_chunked.scale(zeta_to_domain_size - Fr::<G>::one())
         };
 
-        let t_chunked = quotient_poly.chunk_polynomial(zeta_to_srs_len, index.max_poly_size);
+        //~ 32. construct the blinding part of the ft polynomial commitment
+        //~     see https://o1-labs.github.io/mina-book/crypto/plonk/maller_15.html#evaluation-proof-and-blinding-factors
+        let blinding_ft = {
+            let blinding_t = t_comm.1.chunk_blinding(zeta_to_srs_len);
+            let blinding_f = Fr::<G>::zero();
+
+            PolyComm {
+                // blinding_f - Z_H(zeta) * blinding_t
+                unshifted: vec![blinding_f - (zeta_to_domain_size - Fr::<G>::one()) * blinding_t],
+                shifted: None,
+            }
+        };
 
-        let ft: DensePolynomial<Fr<G>> =
-            &f_chunked - &t_chunked.scale(zeta_to_domain_size - Fr::<G>::one());
+        //~ 33. Evaluate the ft polynomial at $\zeta\omega$ only.
         let ft_eval1 = ft.evaluate(&zeta_omega);
 
+        //~ 34. Setup the Fr-Sponge
         let fq_sponge_before_evaluations = fq_sponge.clone();
-        let mut fr_sponge = {
-            let mut s = EFrSponge::new(index.cs.fr_sponge_params.clone());
-            s.absorb(&fq_sponge.digest());
-            s
-        };
-        let p_eval = if public_poly.is_zero() {
+        let mut fr_sponge = EFrSponge::new(index.cs.fr_sponge_params.clone());
+
+        //~ 35. Squeeze the Fq-sponge and absorb the result with the Fr-Sponge.
+        fr_sponge.absorb(&fq_sponge.digest());
+
+        //~ 36. Evaluate the negated public polynomial (if present) at $\zeta$ and $\zeta\omega$.
+        let public_evals = if public_poly.is_zero() {
             [Vec::new(), Vec::new()]
         } else {
             [
                 vec![public_poly.evaluate(&zeta)],
                 vec![public_poly.evaluate(&zeta_omega)],
             ]
         };
+
+        //~ 37. Absorb all the polynomial evaluations in $\zeta$ and $\zeta\omega$:
+        //~     - the public polynomial
+        //~     - z
+        //~     - generic selector
+        //~     - poseidon selector
+        //~     - the 15 register/witness
+        //~     - the 6 sigmas evaluations
```

Comment:
6 or 7? also, continuing on another comment of the review, is there a sigma for each I/O register?

Reply:
6 actually, because one is not evaluated and kept as a commitment